EventFilter/L1TRawToDigi/plugins/implementations_stage2/CaloSummaryUnpacker.cc

```diff
@@ -0,0 +1,48 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "EventFilter/L1TRawToDigi/plugins/UnpackerFactory.h"
+
+#include "L1Trigger/L1TCalorimeter/interface/CaloTools.h"
+
+#include "L1TObjectCollections.h"
+
+#include "CaloSummaryUnpacker.h"
+#include "GTSetup.h"
+
+#include <cmath>
+
+float l1t::stage2::CaloSummaryUnpacker::processBitsToScore(const unsigned int bitsArray[]) {
+  float constructedScore = 0.0;
+  //All bits have been shifted to just left of the decimal point
+  //We need to convert them to float, shift them back to their proper position
+  //And then add them into the total
+  //The proper power is 4(-(bitIndex+1) + numCICADAWords/2)
+  // i.e. shift bitIndex to max out at half the number of CICADA words (indexed at 0) then count down
+  //And we shift by 4 bits a time, hence the factor of 4
+  for (unsigned short bitIndex = 0; bitIndex < numCICADAWords; ++bitIndex) {
+    constructedScore += ((float)bitsArray[bitIndex]) * pow(2.0, 4 * (numCICADAWords / 2 - (bitIndex + 1)));
+  }
+  return constructedScore;
+}
+
+bool l1t::stage2::CaloSummaryUnpacker::unpack(const Block& block, UnpackerCollections* coll) {
+  LogDebug("L1T") << "Block ID  = " << block.header().getID() << " size = " << block.header().getSize();
+  std::cout << "Unpacking CICADA!" << std::endl;
```

Comment:
cout can not be in production code - use LogDebug or so.

Reply:
Ahh, my bad. I know better than that. Forgot the debug statement. Thanks for catching that.

Reply:
Fixed in recent (squashed) commit